src/vm/eventpipeprovider.cpp

```diff
@@ -197,6 +197,30 @@ void EventPipeProvider::InvokeCallback()
     }
     CONTRACTL_END;
 
+    const unsigned int MaxNumberOfBytes = 1024;
+    WCHAR buffer[MaxNumberOfBytes / sizeof(WCHAR)]{};
+    EventFilterDescriptor eventFilterDescriptor{};
+    bool isEventFilterDescriptorInitialized = false;
+
+    if (pFilterData != NULL)
+    {
+        // Why can't we use standard C++ algorithms and containers?
+        const size_t filterDataLength = wcslen(pFilterData) + 1;
+        const size_t filterDataSizeInBytes = filterDataLength * sizeof(WCHAR);
+        const size_t nCharactersToCopy = filterDataSizeInBytes <= sizeof(buffer) ?
+            filterDataLength : sizeof(buffer) / sizeof(WCHAR);
+        wcscpy_s(buffer, nCharactersToCopy, pFilterData);
+
+        for (size_t i = 0; i < nCharactersToCopy; ++i)
+            if (buffer[i] == L'=' || buffer[i] == L';')
+                buffer[i] = L'\0';
+
+        eventFilterDescriptor.Ptr = (ULONGLONG)&buffer[0];
+        eventFilterDescriptor.Size = (ULONG)nCharactersToCopy;
+        eventFilterDescriptor.Type = 0; // TODO: What should the type be?
```

Comment:
This is logically ControllerCommand.Update (defined in EventProvider.cs) which is 0.